### DISCUSSION TITLE
Windows Package Manager v1.0.11451

### DISCUSSION THREAD

**Discussion by denelon** [microsoft/winget-cli#988] [Releases]
Windows Package Manager v1.0

This release includes all of the features and bug fixes from the[ v1.0 milestone](https://github.com/microsoft/winget-cli/milestone/1).
```
The following commands are available:
  install    Installs the given package
  show       Shows information about a package
  source     Manage sources of packages
  search     Find and show basic info of packages
  list       Display installed packages
  upgrade    Upgrades the given package
  uninstall  Uninstalls the given package
  hash       Helper to hash installer files
  validate   Validates a manifest file
  settings   Open settings
  features   Shows the status of experimental features
  export     Exports a list of the installed packages
  import     Installs all the packages in a file
```

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/winget-cli/releases/tag/v1.0.11451'>Windows Package Manager v1.0.11451</a>.</em>

**Comment by jedieaston** [comment]
Hooray! Now Windows has a package manager!

**Reply by youk** [reply]
No, it doesn't. Do you see anything in Windows Store?

**Reply by hoang-himself** [reply]
![](https://i.kym-cdn.com/photos/images/original/001/621/567/bdb.jpg)

**Reply by RMITBLOG** [reply]
congrats @denelon  👍

**Comment by auanasgheps** [comment]
Finally!

To be honest I thought the original milestone was pretty tight but you made it!

**Comment by blai30** [comment]
How do I remove my current installation of winget-cli? I want to receive automatic updates from the Microsoft Store through the App Installer package

**Reply by denelon** [reply]
You should get automatic updates from the store as soon as any "newer" version of the App Installer is published (even if you installed a version from GitHub). There are some variables about whether you are on a Windows Insider build and whether you are in the Windows Package Manager Insider program to determine when the next "newer" version is published.

Do you want to continue getting "Development" builds, or do you just want "stable" releases?

If you want "Development" builds going forward, they will be available to Windows Insiders (DEV channel), and to users who have signed up for the Windows Package Manager Insider program.

If you want "stable" releases (and you have installed 1.0), then you will automatically get the 1.1 upgrade when that is released to "all" users.

**Reply by blai30** [reply]
My current version of winget-cli is `v0.2.3162 Preview` which I installed from GitHub releases a while back. I have not updated it since. I want to receive stable releases from the Microsoft Store from now on. I assume I will have to wait a little bit for the 1.0 release to hit the Microsoft Store?

**Reply by denelon** [reply]
Any version you install from GitHub releases would be updated when a newer version lands in the store (for your signed-in MSA).

If you are running any Windows Insider build, you would get 1.0 via the store.
If you are in the Windows Package Manager Insider group, you would get 1.0 via the store.

In your case, with v0.2 you are not likely on a Windows Insider build or logged into the store with an MSA in the Windows Package Manager Insiders group. So, I would suggest [downloading and installing](https://github.com/microsoft/winget-cli/releases/download/v1.0.11451/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.appxbundle) from the 1.0 release.

When the next stable version releases, you will get the upgrade.

**Reply by auanasgheps** [reply]
Hey, when/how winget will be released officially to the public? 
I always installed it manually from Github, but never joined any Insider group and I would like to get automatic updates as well. 
On new PCs, to get the production version, what should a user do? Docs are not yet updated.

**Reply by denelon** [reply]
We're pausing on the push to "all" Windows 10 1809 and newer machines for our enterprise customers. We want to make sure they have sufficient time to implement Group Policy. 

Once we are shipped in the default App Installer from the store you will receive automatic updates.

You can use https://aka.ms/getwinget to pull down the "latest" build, or come directly to the releases page for any new PC.

**Reply by lone-wolf-akela** [reply]
> If you want "Development" builds going forward, they will be available to Windows Insiders (DEV channel), and to users who have signed up for the Windows Package Manager Insider program.

I have signed up for the Windows Package Manager Insider program. Is there any way to quit the program so I can get the stable version after it is available from Microsoft Store?

**Reply by denelon** [reply]
Yes, there is a form. I need to update the documentation and add the link. We will also include instructions for returning to the current stable version.

**Comment by G1Joshi** [comment]
Eagerly waiting for this 🔥

**Reply by denelon** [reply]
You can [install it now](https://aka.ms/getwinget), and you will continue to get updates automatically from the store.

**Reply by G1Joshi** [reply]
Thank you :)

**Reply by soroshsabz** [reply]
@denelon your link does not work

**Reply by ImJoakim** [reply]
> @denelon your link does not work

Hey @soroshsabz, here's a link to the latest version: [Windows Package Manager v1.0.11692](https://github.com/microsoft/winget-cli/releases/download/v1.0.11692/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle)

**Reply by soroshsabz** [reply]
@ImJoakim thanks for reply, but my question why we cannot install official app from Microsoft store? I think version 1 is stable and distribute over Microsoft Store, Did I correct or not?

thanks

**Reply by ImJoakim** [reply]
> @ImJoakim thanks for reply, but my question why we cannot install official app from Microsoft store? I think version 1 is stable and distribute over Microsoft Store, Did I correct or not?
> 
> thanks

@soroshsabz This discussion message might be useful in regards to your question: https://github.com/microsoft/winget-cli/discussions/988#discussioncomment-793667

I believe the current plan is September.

**Reply by denelon** [reply]
@soroshsabz, thanks, I updated the link to just point to the releases page.

**Reply by ImJoakim** [reply]
> @soroshsabz, thanks, I updated the link to just point to the releases page.

Hey @denelon, I just noticed the https://aka.ms/getwinget link that currently directs you to the latest download link. Perhaps you should change it out with this one (if it gets updated as soon as new release comes out). It might be easier (in my opinion) for users to click on the direct download instead of the release page itself.

**Comment by xdagee** [comment]
awesomeness 😎

**Comment by wasserja** [comment]
When I install this AppX package it is listed as version 1.11.11451.0 when running Get-AppXPackage, but the version number is listed as 1.0.11451. Can you confirm what it is supposed to be?

**Reply by denelon** [reply]
We use version 1.11.114510.0 for the "App Installer" package. The Windows Package Manager ships "inside" that package and reports as 1.0.11451. You have the correct version.

**Reply by denelon** [reply]
You can confirm with `winget --info`

**Reply by wasserja** [reply]
PS C:\> winget --version
v1.0.11451
PS C:\> Get-AppxPackage microsoft.desktopappinstaller


Name              : Microsoft.DesktopAppInstaller
Publisher         : CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US
Architecture      : X64
ResourceId        :
Version           : 1.11.11451.0
PackageFullName   : Microsoft.DesktopAppInstaller_1.11.11451.0_x64__8wekyb3d8bbwe
InstallLocation   : C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_1.11.11451.0_x64__8wekyb3d8bbwe
IsFramework       : False
PackageFamilyName : Microsoft.DesktopAppInstaller_8wekyb3d8bbwe
PublisherId       : 8wekyb3d8bbwe
IsResourcePackage : False
IsBundle          : False
IsDevelopmentMode : False
NonRemovable      : False
Dependencies      : {Microsoft.VCLibs.140.00.UWPDesktop_14.0.29231.0_x64__8wekyb3d8bbwe,
                    Microsoft.DesktopAppInstaller_1.11.11451.0_neutral_split.scale-100_8wekyb3d8bbwe}
IsPartiallyStaged : False
SignatureKind     : Store
Status            : Ok

**Reply by wasserja** [reply]
It's just confusing why there are two different versions reported.

**Reply by denelon** [reply]
It's the result of how we're "Flighting" the Windows Package Manager Insider builds. We use 1.10.x for stable "App Installer" flights in the store. It would have been a bad experience if you were in the "insider" program and a change was made to the "App Installer" package without the Windows Package Manager and it was automatically installed on top of the insider build. We needed to bump the minor version up to prevent that from happening. Once the Windows Package Manager ships with the default App Installer, we may converge the versions back together again.

**Comment by tonypags** [comment]
Congratulations on the release of v1.0!

**Comment by jueane** [comment]
A great experience!

**Comment by silentflight** [comment]
Looks nice! Are installers required to be signed by the listed author/publisher?  If not, what prevents someone from updating say VSCode's manifest with an otherwise non-malicious (passes static analysis) non-VSCode installer and supply a valid hash?

>We leverage SmartScreen, static analysis, SHA256 hash validation and a few other processes to reduce the likelihood of malicious software making its way into the repository and onto your machine

I can infer that this is handled as part of SmartScreen or "other processes" but couldn't find any exact explanation.

**Reply by denelon** [reply]
Installers are not required to be signed. We're still working on the "package ownership" feature, and it's just about complete. That will help in cases where the publisher/ISV wishes to control who makes changes to a manifest. For other community packages, it's a combination of some automation looking at URLs and some manual validation. In the event someone does what you've described, either community would remove it, or one of the project maintainers would remove it. We may also need to look at adding some form of moderation in the future as well.

**Comment by Mental-floss** [comment]
Gee, Microsoft stole yet another great product from someone else. Who saw that coming?

**Reply by scottbeamer** [reply]
What?

**Reply by Gteditor99** [reply]
What do you mean?

**Reply by Mental-floss** [reply]
I've worked in IT since the early 80's and have watched Microsoft pull this stunt repeatedly, or at times acquiring a technology and reneging to pay for it such as with Mosaic. My comment is based on this link below. Microsoft pulled the same crap with a graphics tech telling them they were going to buy them and keeping them from releasing a product until announcing they were purchasing DirectX. https://www.thurrott.com/windows/windows-10/235783/appget-creator-says-microsoft-stole-his-product

**Reply by Masamune3210** [reply]
Cool, I want to see them strongarm a idea, considering the idea of a package manager has been in almost every os other than the ones Microsoft were directly involved with. 

Just because they finally decided to make their own doesn't mean they are trying to steal something, especially something that can't be stolen

**Reply by jedieaston** [reply]
This discussion has already been had, see #353.

**Reply by zerst-au** [reply]
Basically the same as AppGet is not the same as AppGet...

**Comment by na-jakobs** [comment]
Is it possible to install v1.0 of winget on Server2019?

**Reply by denelon** [reply]
We haven't actively been targeting server SKUs just yet. I believe some users have been able to install it. The dependencies seem to be part of the challenge as well as a license file prerequisite. I believe this Issue should lead you to what others have done, and the request for support on server SKUs. Be sure to add your 👍 

https://github.com/microsoft/winget-cli/issues/754

**Comment by aeongdesu** [comment]
👏😎

**Comment by jarmos-san** [comment]
So, now that Winget got it's first major release, is it possible to install zipped binaries? Or is it still under the works at the moment?

Related to #140 

And lastly, I want to thank the original creator of AppGet & the contributors to `winget` for making it possible!

**Reply by denelon** [reply]
#140 is the key to unlocking zipped binaries.

**Comment by raspiduino** [comment]
When will this tool available natively in Windows? 21H2?
I really love this tool!

**Reply by denelon** [reply]
@raspiduino
We are aiming for the September 21' timeframe (not a date commitment). This is a new way to install software on Windows 10 RS5+, so we need to provide enterprises (IT Professionals) enough time to determine their Group Policy configurations and roll them out.

We're thrilled to hear your support for the tool, and we are looking forward to rolling it out to everyone as soon as its appropriate. Be sure to 👍 your favorite features.

**Reply by raspiduino** [reply]
Thanks!

**Comment by Lamparter** [comment]
Woohoo! well done msft!

**Comment by jravi7** [comment]
Is there a way to disable GUI prompts when installing using `winget`? I ask because I would like to use it with Docker on Windows. This leads me to the next question, how can install this tool as part of my Dockerfile setup?

**Reply by denelon** [reply]
@jaykumarark you may specify "silent" for the install. Check the arguments with `winget install -?`. Take a look at #438 for Docker and add your 👍 to help raise priority.

**Reply by jravi7** [reply]
Thanks for the tip on silent installations. And I voted for that issue for Docker. Thanks!

**Comment by capsalad** [comment]
I'm seeing some strange behavior with the upgrade mechanism in Winget v. 1.0.11451.

On my system, `winget upgrade` lists six packages which are apparently not upgraded by `winget upgrade --all`:

```
Name                              Id                                    Version       Available       Source
-------------------------------------------------------------------------------------------------------------
Visual Studio Community 2019      Microsoft.VisualStudio.2019.Community 16.9.31112.23 16.10.31402.337 winget
7-Zip Alpha                       7zip.7zipAlpha                        21.02 alpha   21.02.00.0      winget
Code Writer                       ActiproSoftwareLLC.562882FEEB491      4.2.42.0      Latest          msstore
Python 3.9                        PythonSoftwareFoundation.Python.3.9   3.9.1520.0    Latest          msstore
Microsoft Windows Desktop Runtime Microsoft.dotnetRuntime               3.1.11.29516  5.0.6.30021     winget
Microsoft .NET SDK                Microsoft.dotnet                      5.2.321.22315 5.3.121.27113   winget
```

The two packages (**Code Writer** and **Python 3.9**) sourced from the Microsoft Store may be hitting an oversight bug if the upgrade mechanism's version check is attempting a numeric comparison against the string "Latest" (perhaps the real issue is that the Microsoft Store doesn't reliably supply programmatically-accessible version numbers?).

Something similar appears to be the case for the 7-Zip Alpha package, but in this case the package itself is labelled with the string "21.02 alpha", while the Winget repo uses "21.02.00.0". Of course that comparison will return `False`.

The other three packages are more of a mystery in that the false comparisons appear to be legitimate. In the case of Microsoft's .NET and .NET Runtime packages, there may be a problem with one-to-many relationships: attempts to manually upgrade them result in "Multiple installed packages found matching input criteria. Please refine the input." But even attempts to upgrade them using the packages' GUIDs fail with "No applicable update found."

Finally, we have Visual Studio 2019 Community Edition. The version check looks legitimate, and `winget upgrade Microsoft.VisualStudio.2019.Community` results in "Successfully installed"; but subsequent calls to `winget upgrade` show that the installed version apparently hasn't incremented, and the app itself shows a version stamp of "16.0.31112.23 D16.9". Could this be another issue with a string _vs._ float comparison?

**Comment by denelon** [comment]
@capsalad
>The two packages (Code Writer and Python 3.9) sourced from the Microsoft Store may be hitting an oversight bug if the upgrade mechanism's version check is attempting a numeric comparison against the string "Latest" (perhaps the real issue is that the Microsoft Store doesn't reliably supply programmatically-accessible version numbers?).

That is correct, the store does not provide the version programmatically.

We still have more work to support better matching and upgrade experiences in Add / Remove Programs (ARP). Give you 👍 on #1073 and #1145 to raise priority.
